python/GafferUI/GraphEditor.py

```diff
@@ -231,6 +232,34 @@ def appendEnabledPlugMenuDefinitions( cls, graphEditor, node, menuDefinition ) :
 				}
 			)
 
+			# The plug is hidden as it has no nodule type, rather than via visibility
+			plugVisible = bool( Gaffer.Metadata.value( enabledPlug, "nodule:type" ) )
+			menuDefinition.append(
+				"/Show Enabled Plug",
+				{
+					# If we weakref enabledPlug, it is dead by the time we come to use it (?)
+					"command" : functools.partial( GafferUI.GraphEditor.__exposeEnabledPlug, weakref.ref(node), not plugVisible ),
```

Comment:
@johnhaddon Must be missing something obvious, but wasn't sure why taking a `weakref.ref` to `enabledPlug` would always be dead when the menu item was selected.  I thought plugs were stable along with the lifetime of the node?

Reply:
The lifetime of the Python object representing the plug isn't tied to the lifetime of the C++ object in the way that you want. The Python object owns a reference to the C++ object, but not the other way around. You're just getting lucky with the node, because when we bind a class so that it can be inherited in Python, there _is_ a two way relationship.

But why do you want to use a weak reference here at all?

Reply:
Ahh cool, thanks. Makes sense. Ahh yeah - I guess as we destroy the popup when it closes we don't need a `weakref` do we.